ci/brew-bootstrap.inc.sh

```diff
@@ -4,18 +4,34 @@ set -euo pipefail
 SUPPORT_FIRECLOUD_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
 source ${SUPPORT_FIRECLOUD_DIR}/sh/common.inc.sh
 
+HAS_BREW=$(brew info >/dev/null 2>&1 ; echo $?)
+
 case $(uname -s) in
     Darwin)
-        echo_do "brew: Installing homebrew..."
-        </dev/null ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
-        echo_done
+        if [ "$HAS_BREW" != "0" ]
+        then
+            echo_do "brew: Installing homebrew..."
+            </dev/null ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+            echo_done
+        fi
 
         CI_CACHE_HOMEBREW_PREFIX=~/.homebrew
         ;;
     Linux)
-        echo_do "brew: Installing linuxbrew..."
-        </dev/null sh -c "$(curl -fsSL https://raw.githubusercontent.com/Linuxbrew/install/master/install.sh)"
-        echo_done
+        if [ "$HAS_BREW" != "0" ]
+        then
+            echo_do "brew: Installing linuxbrew..."
+            if [ "$SF_USE_SUDO" = "false" ]
+            then
+                echo_do "brew: Installing without sudo into $HOME/.linuxbrew"
```

Comment:
you can look in the travis output, and in `bin/ci-echo` for the implementation and behaviour of `echo_do` and `echo_done` but in short we create log groups that travis and other CIs fold automatically

so if you use a `echo_do`, use a `echo_done` as well, otherwise the folding, the automatic timing of a coding block, etc gets off